GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1084,7 +1089,7 @@ private void UpdatePlaceholders()
             using (ITracer activity = this.context.Tracer.StartActivity("UpdatePlaceholders", EventLevel.Informational, metadata))
             {
                 ConcurrentHashSet<string> folderPlaceholdersToKeep = new ConcurrentHashSet<string>();
-                ConcurrentBag<PlaceholderListDatabase.PlaceholderData> updatedPlaceholderList = new ConcurrentBag<PlaceholderListDatabase.PlaceholderData>();
+                ConcurrentDictionary<string, PlaceholderListDatabase.PlaceholderData> updatedPlaceholderList = new ConcurrentDictionary<string, PlaceholderListDatabase.PlaceholderData>(StringComparer.Ordinal);

```

Comment:
> Also how much overhead does adding the string key add? 

Let me measure that.  If it's significant we might need to further split this code out between Mac and Windows (because we do need the dictionary on Mac).

> Should this be a ConcurrentDictionary<string, string> and we build the PlaceholderListDatabase.PlaceholderData when calling WriteAllEntriesAndFlush?

That's a good idea, let me give that a try before making any perf measurements.

Reply:
> Should this be a ConcurrentDictionary<string, string> and we build the PlaceholderListDatabase.PlaceholderData when calling WriteAllEntriesAndFlush

I'm actually going to start with the perf overhead (starting that testing this morning) and see how things look before and after this change.  We're using the same string instances in both the `ConcurrentDictionary` and the `PlaceholderData` and so the memory overhead is just an extra reference.

Reply:
> Should this be a ConcurrentDictionary<string, string> and we build the PlaceholderListDatabase.PlaceholderData when calling WriteAllEntriesAndFlush?

Also, to confirm, what was the main concern here? Perf or memory usage?